packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +58,783 @@ export class ModelMesh extends Mesh {
     super(engine);
     this.name = name;
   }
+
+  /**
+   * Upload Mesh Data to the graphics API.
+   * @param noLongerAccessible - Whether to access data later. If true, you'll never access data anymore (free memory cache)
+   */
+  uploadMeshData(noLongerAccessible: boolean): void {
+    const { _indicesArray, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      const vertices = new Float32Array(elementCount * this.vertexCount);
+      this._verticesArray = vertices;
+      this._vertexChangeFlag = ValueChanged.All;
+      this._resetVertexArrayData(vertices);
+      this._vertexBuffer = new Buffer(
+        this._engine,
+        BufferBindFlag.VertexBuffer,
+        vertices,
+        noLongerAccessible ? BufferUsage.Static : BufferUsage.Dynamic
+      );
+      this.setVertexElements(vertexElements);
+      this.setVertexBufferBinding(new VertexBufferBinding(this._vertexBuffer, elementCount * 4));
+      this._vertexSlotChanged = false;
+      this._vertexCountChanged = false;
+    } else {
+      // TODO reset value
+      const verticesArray = this._verticesArray;
+      this._resetVertexArrayData(verticesArray);
+      this._vertexBuffer!.setData(verticesArray);
+    }
+
+    if (_indicesArray) {
+      if (!this._indexBuffer) {
+        this._indexBuffer = new Buffer(this._engine, BufferBindFlag.IndexBuffer, _indicesArray);
+        this.setIndexBufferBinding(new IndexBufferBinding(this._indexBuffer, this._indicesFormat));
```

Comment:
Can use "this.setIndexBufferBinding(this._indexBuffer, this._indicesFormat);" to simplify code